### PR TITLE
Delete Expr layer.

### PR DESCRIPTION
**Pull request by aniemetz**

### CODE REVIEW

**Review by ajreynol — CHANGES_REQUESTED** [review]
It looks like the is a debug CI failure.

I have one comment, otherwise this looks great.

File: src/expr/node.h [hunk 1]
```diff
@@ -43,6 +43,7 @@
 
 namespace CVC4 {
 
+class Type;
```
[the remark this answers is not in the document]
Comment (mpreiner) [reply src/expr/node.h:46]: It shouldn't be since `Type` gets deleted.

**Review by mpreiner — APPROVED** [review]
LGTM except for Andy's comment.

File: src/expr/node.h (hunk 1 above)
[the remark this answers is not in the document]
Comment (aniemetz) [reply src/expr/node.h:46]: Good catch! Yes, it isn't anymore, it's a leftover from an intermediate state.

**Review by ajreynol — APPROVED** [review]

**Pull request merged**